src/control/cmd/daos_agent/procmon.go

```diff
@@ -341,6 +342,52 @@ func (p *procMon) handleRequests(ctx context.Context) {
 
 // startMonitoring is the main driver which starts the process monitor. The
 // passed in context is used to terminate all monitoring in the event of shutdown.
-func (p *procMon) startMonitoring(ctx context.Context) {
+func (p *procMon) startMonitoring(ctx context.Context, cleanOnStart bool) {
+	if cleanOnStart {
+		p.cleanupServerHandles(ctx)
+	}
 	go p.handleRequests(ctx)
 }
+
+// cleanupServerHandles can be run to revoke all pool handles associated with a given machine/host.
+// DAOS server will be instructed to cleanup handles associated with the source machine/host.
+func (p *procMon) cleanupServerHandles(ctx context.Context) {
+	machineName, err := auth.GetMachineName()
+	if err != nil {
+		p.log.Errorf("hostname lookup: %s, cannot cleanup handles", err)
+		return
+	}
+	if machineName == "" {
+		p.log.Errorf("empty machine name is invalid, cannot cleanup handles", err)
+		return
+	}
+
+	req := &control.SystemCleanupReq{Machine: machineName}
+	req.SetSystem(p.systemName)
+
+	msg := fmt.Sprintf("machine %q", machineName)
+	resp, err := control.SystemCleanup(ctx, p.ctlInvoker, req)
+	if err != nil {
+		p.log.Errorf("%s: failed to run system cleanup: %s", msg, err)
+		return
+	}
+	err = resp.Errors()
+	if err != nil {
+		p.log.Errorf("%s: system cleanup ran with error: %s", msg, err)
+		return
+	}
+
+	msg = fmt.Sprintf("Running system cleanup on this %s", msg)
+	if len(resp.Results) == 0 {
+		p.log.Debugf("%s: no pool handles revoked", msg)
+		return
+	}
+	msgRvkd := ""
+	for i, r := range resp.Results {
+		msgRvkd = fmt.Sprintf("%s%d handles revoked for pool %q", msgRvkd, r.Count, r.PoolID)
+		if i != len(resp.Results)-1 {
+			msgRvkd += ", "
+		}
+	}
+	p.log.Infof("%s: %s", msg, msgRvkd)
```

Comment:
Just a suggestion, but maybe this log message could be made a bit less verbose, something like "poolID (count)" for each pool. The overall message could look something like:
`Pool handles revoked: pool1 (3), pool5 (10), pool6 (1)`

Reply:
This indeed, could be helpful for support tasks.

Reply:
Yes this is much better, thanks. Done